agent/eventhandler/task_handler.go

```diff
@@ -334,7 +342,7 @@ func (handler *TaskHandler) submitTaskEvents(taskEvents *taskSendableEvents, cli
 		// If we looped back up here, we successfully submitted an event, but
 		// we haven't emptied the list so we should keep submitting
 		backoff.Reset()
-		retry.RetryWithBackoff(backoff, func() error {
+		retry.RetryWithBackoffForTaskHandler(handler.cfg, handler.eventFlowController, taskARN, handler.delay, backoff, func() error {
```

Comment:
Maybe I'm oversimplifying, but why does this flow controller need to exist per-task? When connection is resumed it always applies to all tasks doesn't it? Could we do this with a single global mutex/channel/atomic to indicate when connection is established or not?

Reply:
Not really sure how it would work with one channel... Afaik a channel will let me do 1:1 communication between two threads, but since there are many threads executing the `RetryWithBackoffForTaskHandler` asynchronously, I need a 1:n communication. In this scenario, if I have multiple threads receiving on a single channel, only one of them will receive the message that I send on the channel